banyan/src/forest/mod.rs

```diff
@@ -5,33 +5,67 @@ use anyhow::Result;
 use core::{fmt::Debug, hash::Hash, iter::FromIterator, marker::PhantomData, ops::Range};
 use futures::future::BoxFuture;
 use libipld::cbor::DagCbor;
-use lru::LruCache;
+use parking_lot::Mutex;
 use rand::RngCore;
-use std::{
-    fmt::Display,
-    sync::{Arc, Mutex},
-};
+use std::{fmt::Display, num::NonZeroUsize, sync::Arc};
+use weight_cache::{Weigheable, WeightCache};
 mod read;
 mod stream;
 mod write;
 pub(crate) use read::ForestIter;
 
 pub type FutureResult<'a, T> = BoxFuture<'a, Result<T>>;
 
+impl<T: TreeTypes> Weigheable for Branch<T> {
```

Comment:
I think this trait should be called `Weighable`. See https://en.wiktionary.org/wiki/weighable